File: src/0011.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        l, r = 0, len(height) - 1
        ans = 0
        while l < r:
            ans = max(ans, (r-l)*min(height[l], height[r]))
            if height[r] > height[l]:
                l += 1
            else:
                r -= 1
        return ans

File: src/0011.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0, front = 0, end = height.size() - 1;
        while(end > front) {
            ans = max(ans, (end-front)*min(height[front], height[end]));
            height[front] < height[end] ? front += 1 : end -= 1;
        }
        return ans;
    }
};
